derp/derp_server.go

```diff
@@ -329,15 +329,16 @@ func NewServer(privateKey key.NodePrivate, logf logger.Logf) *Server {
 	s.initMetacert()
 	s.packetsRecvDisco = s.packetsRecvByKind.Get("disco")
 	s.packetsRecvOther = s.packetsRecvByKind.Get("other")
-	s.packetsDroppedReasonCounters = []*expvar.Int{
-		s.packetsDroppedReason.Get("unknown_dest"),
-		s.packetsDroppedReason.Get("unknown_dest_on_fwd"),
-		s.packetsDroppedReason.Get("gone_disconnected"),
-		s.packetsDroppedReason.Get("gone_not_here"),
-		s.packetsDroppedReason.Get("queue_head"),
-		s.packetsDroppedReason.Get("queue_tail"),
-		s.packetsDroppedReason.Get("write_error"),
-	}
+
+	s.packetsDroppedReasonCounters = make([]*expvar.Int, 7)
+	s.packetsDroppedReasonCounters[dropReasonUnknownDest] = s.packetsDroppedReason.Get("unknown_dest")
+	s.packetsDroppedReasonCounters[dropReasonUnknownDestOnFwd] = s.packetsDroppedReason.Get("unknown_dest_on_fwd")
+	s.packetsDroppedReasonCounters[dropReasonGoneDisconnected] = s.packetsDroppedReason.Get("gone_disconnected")
+	s.packetsDroppedReasonCounters[dropReasonQueueHead] = s.packetsDroppedReason.Get("queue_head")
+	s.packetsDroppedReasonCounters[dropReasonQueueTail] = s.packetsDroppedReason.Get("queue_tail")
+	s.packetsDroppedReasonCounters[dropReasonWriteError] = s.packetsDroppedReason.Get("write_error")
+	s.packetsDroppedReasonCounters[dropReasonDupClient] = s.packetsDroppedReason.Get("dup_client")
```

Comment:
Looks like dup_client was the only missing one and all the others were fine? One was just renamed, but doesn't look like anything was reordered or off-by-one?

Reply:
No, `queue_head`, `queue_tail` and `write_error` were off by one, and `dup_client` was not included in the stats.  E.g. I saw in stats:

```
"counter_packets_dropped_reason": {
  "gone_disconnected": 0,
  "gone_not_here": 20615, 
  "queue_head": 0, 
  "queue_tail": 0, 
  "unknown_dest": 26, 
  "unknown_dest_on_fwd": 0, 
  "write_error": 0
}
```

...that is, `dropReasonQueueHead` was getting tracked as `gone_not_here` in the stats.

Reply:
Ah, thanks. Looks like a regression from 74eb99aed175398d9085f54cecb2a64797e93c40.